### PR TITLE
Sources - Ticket Tailor

### LINKED ISSUE #3811: Ticket Tailor
### App
Ticket Tailor

### Triggers
* New Attendee
* New Ticket Order

### PR DESCRIPTION
**Pull request by vellames**
Closes #3811

Obs: One source can handle all ACs of #3811 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4704"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/ticket_tailor/sources/new-action/new-action.mjs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          key,
+          value,
+        ] = currentValue.split("=");
+        acc[key] = value;
+        return acc;
+      }, {});
+    },
+    checkHmac(bodyRaw, timestamp, hmac) {
+      if (!this.sharedSecret) {
+        console.log("No shared secret configured. Skipping HMAC check.");
```
Comment (luancazarine) [inline components/ticket_tailor/sources/new-action/new-action.mjs:42]: is this right? isn't it better to throw an error here?

**Review by luancazarine — COMMENTED** [review]
Hi @vellames, I've added a minor question. LGTM!

File: components/ticket_tailor/sources/new-action/new-action.mjs (hunk 1 above)
Comment (vellames) [reply components/ticket_tailor/sources/new-action/new-action.mjs:42]: Not really. I think this is a user choice. Even though I highly recommend the user to use a secret. It is not mandatory

**Comment by DilanAthukorala** [comment]
@vellames Following issues observed;

1. Typo in description - **webhook** 
image


2. Events are not emitted without `shared secret` , please update the prop to mandatory if this is a mandatory field
image
image

**Comment by vellames** [comment]
@DilanAthukorala Fixed. Key it is not mandatory, it was a bug

**Comment by DilanAthukorala** [comment]
> @DilanAthukorala Fixed. Key it is not mandatory, it was a bug

@vellames Still it's not triggering events when share secret is empty

**Comment by vellames** [comment]
@DilanAthukorala Strange, it is working for me. I will ping you

**Comment by DilanAthukorala** [comment]
Verification success, Ready for Release!

**Review by luancazarine — APPROVED** [review]

**Pull request merged**